qiskit_addon_obp/utils/simplify.py

```diff
@@ -64,6 +64,12 @@ class OperatorBudget:
     simplify: bool = True
     """A flag denoting whether to call :func:`simplify` at every iteration."""
 
+    atol: float = 1e-8
+    """Absolute tolerance for checking if coefficients are zero."""
+
+    rtol: float = 1e-8
+    """Relative tolerance for checking if coefficients are zero."""
```

Comment:
Why are these defaults different to the ones of `SparsePauliOp.atol` and `.rtol`? This means that this change will imply a change in behavior after merging this PR (although a very minimal one).
Would it not make more sense to:
- either use the same default values
- or set these defaults to `None` documenting that they fall back to the `SparsePauliOp` values

I prefer the latter because it means a better tie-in into Qiskit. It is also in-line with the `atol` and `rtol` arguments of the `simplify` function below.
Furthermore, if a user changed the `SparsePauliOp.atol` and `.rtol` values, that won't take affect because these settings here take precedence, which is not ideal I would say.